docs/docs/behaviors/macros.md

```diff
@@ -0,0 +1,136 @@
+---
+title: Macro Behavior
+sidebar_label: Macros
+---
+
+## Summary
+
+The macro behavior allows configuring a list of other behaviors to invoke
+when the macro is pressed, including support for completing part of the binding list
+once the macro is released.
+
+## Macro Definition
+
+Each macro you want to use in your keymap gets defined first, then bound in your keymap.
+
+A macro definition looks like:
+
+```
+/ {
+    macros {
+        zed_em_kay: zmk_macro {
+        label = "ZMK_MAC";
+        compatible = "zmk,behavior-macro";
```

Comment:
I've been looking at these instructions to make sure I understood how to define macros. For the most part they are clear, and I had no difficulty. A couple of questions, though:

1. Must this `compatible` line (and perhaps the following `#binding-cells` line) occur inside each particular macro, or can in be placed outside of the individual macros, but still under `macro { ... }` (as is true with combos)? That is, is the following permissible?---

        / {
            macros {
                compatible = "zmk,behavior-macro";
                #binding-cells = <0>;
                zed_em_kay: zmk_macro {
                    label = "ZMK_MAC";
                    bindings =
                        ...
                };
                kay_em_zed: kmz_macro {
                    label = "KMZ_MAC";
                    bindings =
                        ...
                };
            };
        };

2. I'm uncertain what the point of the `label` item is or whether it matters that it be assigned to anything in particular or just any unique string. (Does it have to be all caps, for example?) From the perspective of a user, that `label` item seems superfluous, and might just be made identical to the "node label".

Reply:
Thanks for reviewing!

1. Not easily. At a minimum, the `binding-cells` still needs to be on the node. The compatible we could *maybe* get away w/ moving. Let me ponder more.
2.The label needs to be unique, it's used internally by our Zephyr RTOS... But perhaps we could auto-generate it...

Reply:
Sadly, it really is required. We can definitely consider a C macro to simplify this.

Reply:
Hmm... Thoughts on using this macro to simplify defining them:

```
		ZMK_MACRO(
			dual_sequence_macro,
			wait-ms = <10>;
			bindings
				= <&macro_press &kp LALT>
				, <&macro_tap>
				, <&kp TAB>
				, <&macro_pause_for_release>
				, <&macro_release &kp LALT>
				;
		)
```

Reply:
@okke-formsma @joelspadin ^

Reply:
I haven't documented it yet, but I've added this macro now, if folks are happy with it, I'll add a note at the end of the docs w/ some examples of using the macro to simplify defining them.

Reply:
As noted on Discord, I find it weird that it appears as if the first line is delimited with a comma but all others with semicolons. Maybe it would be less confusing if you formatted these like

```c
ZMK_MACRO(abc_macro,
    wait-ms = <10>;
    tap-ms = <20>;
    bindings = <&kp A &kp B &kp C>;
)
```
to differentiate the macro name from its contents?

Reply:
I updated these, and noted this in the docs covering this. Marking resolved.